internal/cluster/distribution/eks/eksadapter/node_pool_manager.go

```diff
@@ -123,45 +126,91 @@ func generateNodePoolStackName(clusterName string, poolName string) string {
 }
 
 // ListNodePools lists node pools from a cluster.
-func (n nodePoolManager) ListNodePools(ctx context.Context, cluster cluster.Cluster, nodePoolNames []string) ([]eks.NodePool, error) {
-	clusterClient, err := n.dynamicClientFactory.FromSecret(ctx, cluster.ConfigSecretID.String())
+func (n nodePoolManager) ListNodePools(
+	ctx context.Context,
+	c cluster.Cluster,
+	existingNodePools map[string]eks.ExistingNodePool,
+) (nodePools []eks.NodePool, err error) {
+	if c.ConfigSecretID.ResourceID == "" || // Note: cluster is being created or errorred before k8s secret would be available.
+		c.Status == cluster.Deleting {
+		return nil, cluster.NotReadyError{
+			OrganizationID: c.OrganizationID,
+			ID:             c.ID,
+			Name:           c.Name,
+		}
+	}
+
+	clusterClient, err := n.dynamicClientFactory.FromSecret(ctx, c.ConfigSecretID.String())
 	if err != nil {
-		return nil, errors.WrapWithDetails(err, "creating dynamic Kubernetes client factory failed", "cluster", cluster)
+		return nil, errors.WrapWithDetails(err, "creating dynamic Kubernetes client factory failed",
+			"OrganizationID", c.OrganizationID,
```

Comment:
Camel case is preferred at the moment and is/should be used: `organizationId`. Different casing can make searching the logs harder.

Reply:
I agree with a camelCase, this was a copy-paste typo (copied the field name).

I saw a very mixed bag of initialism though.
(The representation is non-exhaustive, the relative ratio felt around 40-60%.)

Upper case initialism organization ID (and cluster ID):
https://github.com/banzaicloud/pipeline/blob/master/internal/cluster/clusteradapter/gorm_clusters.go#L109
https://github.com/banzaicloud/pipeline/blob/master/internal/cluster/distribution/eks/eksprovider/driver/cluster_creator.go#L541
https://github.com/banzaicloud/pipeline/blob/master/internal/cluster/distribution/eks/eksprovider/driver/cluster_deleter.go#L96
https://github.com/banzaicloud/pipeline/blob/master/internal/cluster/distribution/eks/eksprovider/driver/cluster_updater.go#L342
https://github.com/banzaicloud/pipeline/blob/master/internal/helm/envresolver.go#L98

Mixed case initialism:
https://github.com/banzaicloud/pipeline/blob/master/internal/common/commonadapter/secret.go#L88
https://github.com/banzaicloud/pipeline/blob/master/internal/helm/charter.go#L153
https://github.com/banzaicloud/pipeline/blob/master/internal/helm/errors.go#L70

I'm going to update the initials in the changed code to mixed as you suggested, just wanted to let you know it's already very diverse.

Reply:
[Fixed](https://github.com/banzaicloud/pipeline/compare/a66ed0a2166a67445eaa5df4ba76f2191ecc379f..8bcf793d712a07b312bfd8dda9bf7b763367679a).